lib/Jit/LLILCJit.cpp

```diff
@@ -394,3 +395,107 @@ void __cdecl LLILCJit::fatal(int Errnum, ...) {
 void LLILCJit::signalHandler(void *Cookie) {
   // do nothing
 }
+
+template <typename ObjSetT, typename LoadResult>
+void ObjectLoadListener::operator()(llvm::orc::ObjectLinkingLayerBase::ObjSetHandleT ObjHandles,
+                const ObjSetT &Objs, const LoadResult &LoadedObjInfos) {
+  int I = 0;
+  for (const std::unique_ptr<object::ObjectFile> &PObj : Objs) {
```

Comment:
Since this is written as implementing a contract where the pointer kind (and collection type) of the `ObjectFile` pointer collection is (are) abstracted away in the type parameter, seems like you may as well use `const auto &PObj` instead here, to accommodate other pointer kinds.